### DISCUSSION TITLE
Update `$schema` when Biome is updated

### DISCUSSION THREAD

**Discussion by wojtekmaj** [biomejs/biome#3850] [Ideas]
Consider the situation:

A developer sets up `biome.json`, specifying `"$schema": "https://biomejs.dev/schemas/1.8.0/schema.json",` at the top, perhaps following the guidelines, not giving much thought about it.

Biome 1.9.0 comes along, with new, undoubtedly cool features.

The developer would like to use some of these new features and tries to enable them in biome.json. They are met with "Found an unknown key" or similar in their IDE. Unless they know what `$schema` does, they could be puzzled about why their now has errors, despite Biome being fully up to date. 

Solution:

When Biome runs, check if Biome version matches `$schema`. If not, update `$schema` on the fly.

**Comment by Conaclos** [comment]
We recommend running `biome migrate` after updating Biome. The command updates `$schema` and more.

**Comment by minht11** [comment]
Another option, if you are using node_modules, you can point it to local version like this:
`"$schema": "./node_modules/@biomejs/biome/configuration_schema.json"`

**Comment by Conaclos** [comment]
Users [keep reporting](https://github.com/biomejs/biome/issues/3875#issuecomment-2384766168) issues with the schema because they don't update it.

I wonder if we could emit a warning that advise running `biome migrate --write` when we detect that the version of the schema doesn't match the current version of Biome. We could ignore schema that are not `https://biomejs.dev/schemas/<version>/schema.json` (such as `./node_modules/@biomejs/biome/configuration_schema.json`).

**Reply by ematipico** [reply]
That should be very easy, and we could mark the feature as a good first issue.  Wanna create an issue for that?

**Reply by Conaclos** [reply]
#4147
